framework/sysdeps/unix/child_debug.cpp

```diff
@@ -367,25 +367,43 @@ static bool check_gdb_available()
     return false;
 }
 
-static void create_crash_pipe(int xsave_size)
+static bool create_crash_pipe(int xsave_size)
 {
     int socktype = SOCK_DGRAM;
 #ifdef SOCK_CLOEXEC
     socktype |= SOCK_CLOEXEC;
 #endif
     if (socketpair(AF_UNIX, socktype, 0, crashpipe) == -1)
-        return;
+        return false;
 
     // set the buffer sizes
     xsave_size += sizeof(CrashContext::Fixed) + sizeof(CrashContext::mc);
+    xsave_size += 256;      // add some headroom; the Linux kernel subtracts 32
     xsave_size = ROUND_UP_TO(xsave_size, 1024U);
-    setsockopt(crashpipe[CrashPipeParent], SOL_SOCKET, SO_RCVBUF, &xsave_size, sizeof(xsave_size));
-    setsockopt(crashpipe[CrashPipeChild], SOL_SOCKET, SO_SNDBUF, &xsave_size, sizeof(xsave_size));
+
+    int rcvbuf;
+    socklen_t size = sizeof(rcvbuf);
+    if (getsockopt(crashpipe[CrashPipeParent], SOL_SOCKET, SO_RCVBUF, &rcvbuf, &size) == 0) {
+        if (rcvbuf >= xsave_size)
+            xsave_size = 0;
+    }
+    if (xsave_size) {
+        bool ok = setsockopt(crashpipe[CrashPipeChild], SOL_SOCKET, SO_SNDBUF, &xsave_size, sizeof(xsave_size)) == 0;
+        ok = ok && setsockopt(crashpipe[CrashPipeParent], SOL_SOCKET, SO_RCVBUF, &xsave_size, sizeof(xsave_size)) == 0;
```

Comment:
creative but... really?

Reply:
The `if` line was too long, so I split.

I don't think this will EVER fail, at least on Linux. If you can't ask for a measly 10kB skb, then you've got bigger problems.